### PR TITLE
doc: op queue and mclock related options

### PR DESCRIPTION
**Pull request by ivancich**
Additionally makes sure when unexpected values are provided for config
options osd_op_queue or osd_op_queue_cut_off that the default values
are assigned.

Signed-off-by: J. Eric Ivancich <ivancich@redhat.com>

### CODE REVIEW

File: src/common/options.cc [hunk 1]
```diff
@@ -1881,71 +1881,315 @@ std::vector<Option> global_options = {
...
   .set_default("wpq")
-  .set_description(""),
+  .set_description("which operation queue algorithm to use; can be one of five values -- 'wpq', 'prioritized', 'mclock_opclass', 'mclock_client', and 'debug_random'")
+  .add_service("osd")
+  .add_see_also("osd_op_queue_cut_off"),
 
   Option("osd_op_queue_cut_off", Option::TYPE_STR, Option::LEVEL_ADVANCED)
   .set_default("low")
-  .set_description(""),
+  .set_description("the threshold between high priority ops that use strict priority ordering and low priority ops that use a fairness algorithm that may or may not incorporate prioirty; can be one of three values -- 'low', 'high', and 'debug_random'")
```
Comment (Yan-waller) [inline src/common/options.cc]: how about using `.set_long_description()` here and adding some brief description to `.set_description()`? 
others ditto.

File: src/common/options.cc [hunk 2]
```diff
@@ -1881,71 +1881,315 @@ std::vector<Option> global_options = {
 
   Option("osd_op_queue", Option::TYPE_STR, Option::LEVEL_ADVANCED)
   .set_default("wpq")
-  .set_description(""),
+  .set_description("which operation queue algorithm to use; can be one of five values -- 'wpq', 'prioritized', 'mclock_opclass', 'mclock_client', and 'debug_random'")
```
Comment (tchaikov) [inline src/common/options.cc]: how about using `set_enum_allowed()` to enforce the constraint?

File: src/common/options.cc (hunk 1 above)
Comment (ivancich) [inline src/common/options.cc]: So I did that. I looked through options.h to see if there was any guidance about how long a description (not long description) would ideally be and couldn't find anything. Any thoughts?

File: src/common/options.cc (hunk 2 above)
Comment (ivancich) [inline src/common/options.cc]: Good idea. Done.

File: src/common/options.cc (hunk 1 above)
Comment (Yan-waller) [inline src/common/options.cc]: I'm not very sure, but from my point, `set_descriptione()` is used to describe what it is, while `set_long_description()` is used to describe somethings more details, such as  the optional parameters and how to use it, etc..

**Comment by ivancich** [comment]
jenkins retest this please

File: src/common/options.cc [hunk 3]
```diff
@@ -1881,71 +1881,319 @@ std::vector<Option> global_options = {
...
+  .set_description("which operation queue algorithm to use; mclock_opclass and mclock_client are currently experimental")
+  .add_service("osd")
+  .add_see_also("osd_op_queue_cut_off"),
 
   Option("osd_op_queue_cut_off", Option::TYPE_STR, Option::LEVEL_ADVANCED)
   .set_default("low")
-  .set_description(""),
+  .set_enum_allowed( { "low", "high", "debug_random" } )
+  .set_description("the threshold between high priority ops and low priority ops")
+  .set_long_description("the threshold between high priority ops that use strict priority ordering and low priority ops that use a fairness algorithm that may or may not incorporate prioirty")
```
Comment (jdurgin) [inline src/common/options.cc]: typo in the last 'priority'

File: src/common/options.cc [hunk 4]
```diff
@@ -1881,71 +1881,319 @@ std::vector<Option> global_options = {
 
   Option("osd_op_queue", Option::TYPE_STR, Option::LEVEL_ADVANCED)
   .set_default("wpq")
-  .set_description(""),
+  .set_enum_allowed( { "wpq", "prioritized", "mclock_opclass", "mclock_client", "debug_random" } )
+  .set_description("which operation queue algorithm to use")
+  .set_description("which operation queue algorithm to use; mclock_opclass and mclock_client are currently experimental")
```
Comment (tchaikov) [inline src/common/options.cc]: this should be `set_long_description()`,  i guess? and i am not sure if it helps to repeat the short version of description in the long description.

Comment (ivancich) [inline src/common/options.cc]: I'd like to see some overall guidance so this documentation is consistent throughout ceph. Max length for description. Whether long description should contain description.

Comment (tchaikov) [inline src/common/options.cc]: @ivancich does [this](http://docs.ceph.com/docs/master/dev/config/#description-and-long-description) help?

**Comment by ivancich** [comment]
I just rebased it on master to get it past merge conflicts.

File: src/common/options.cc [hunk 5]
```diff
@@ -1881,71 +1881,334 @@ std::vector<Option> get_global_options() {
 
     Option("osd_op_queue", Option::TYPE_STR, Option::LEVEL_ADVANCED)
     .set_default("wpq")
-    .set_description(""),
+    .set_enum_allowed( { "wpq", "prioritized", "mclock_opclass", "mclock_client", "debug_random" } )
+    .set_description("which operation queue algorithm to use")
+    .set_long_description("which operation queue algorithm to use; mclock_opclass and mclock_client are currently experimental")
+    .add_service("osd")
```
Comment (liewegas) [inline src/common/options.cc]: these add_service() lines aren't needed now if it's in teh osd section

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-08-02_10:30:38-rados-wip-kefu-testing-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2017-08-01_17:41:46-rados-wip-kefu-testing-distro-basic-mira/

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-08-04_23:16:58-rados-wip-sage-testing-20170804b-distro-basic-smithi/

**Pull request merged**